File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/dagger/modules/EvaluationModule.java
package se.gustavkarlsson.aurora_notifier.android.dagger.modules;

import dagger.Binds;
import dagger.Module;
import dagger.Reusable;
import se.gustavkarlsson.aurora_notifier.android.evaluation.ChanceEvaluator;
import se.gustavkarlsson.aurora_notifier.android.evaluation.DarknessEvaluator;
import se.gustavkarlsson.aurora_notifier.android.evaluation.GeomagActivityEvaluator;
import se.gustavkarlsson.aurora_notifier.android.evaluation.GeomagLocationEvaluator;
import se.gustavkarlsson.aurora_notifier.android.evaluation.VisibilityEvaluator;
import se.gustavkarlsson.aurora_notifier.android.models.factors.Darkness;
import se.gustavkarlsson.aurora_notifier.android.models.factors.GeomagActivity;
import se.gustavkarlsson.aurora_notifier.android.models.factors.GeomagLocation;
import se.gustavkarlsson.aurora_notifier.android.models.factors.Visibility;

@Module
public abstract class EvaluationModule {

	@Binds
	@Reusable
	abstract ChanceEvaluator<GeomagActivity> bindGeomagActivityEvaluator(GeomagActivityEvaluator evaluator);

	@Binds
	@Reusable
	abstract ChanceEvaluator<GeomagLocation> bindGeomagLocationEvaluator(GeomagLocationEvaluator evaluator);

	@Binds
	@Reusable
	abstract ChanceEvaluator<Visibility> bindVisibilityEvaluator(VisibilityEvaluator evaluator);

	@Binds
	@Reusable
	abstract ChanceEvaluator<Darkness> bindDarknessEvaluator(DarknessEvaluator evaluator);

}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/di/modules/EvaluationModule.kt
package se.gustavkarlsson.skylight.android.di.modules

import se.gustavkarlsson.skylight.android.entities.*
import se.gustavkarlsson.skylight.android.services.ChanceEvaluator

interface EvaluationModule {
    val kpIndexEvaluator: ChanceEvaluator<KpIndex>
    val geomagLocationEvaluator: ChanceEvaluator<GeomagLocation>
    val visibilityEvaluator: ChanceEvaluator<Visibility>
    val darknessEvaluator: ChanceEvaluator<Darkness>
    val auroraReportEvaluator: ChanceEvaluator<AuroraReport>
}
